SimG4Core/Generators/src/Generator.cc

```diff
@@ -275,7 +275,8 @@ void Generator::HepMC2G4(const HepMC::GenEvent * evt_orig, G4Event * g4evt)
 	decay_length = std::sqrt((x2-x1)*(x2-x1)+(y2-y1)*(y2-y1)+(z2-z1)*(z2-z1));
       } 
 
-      bool toBeAdded = false;
+      bool toBeAdded = (fFiductialCuts) ? false : true;
```

Comment:
@civanch with the default g4SimHits settings this should preserve the standard behaviour...

Reply:
@fabiocos , no differences in all tests is observed. Normally we have eta/p cuts, so the code behaves in the same way as before.